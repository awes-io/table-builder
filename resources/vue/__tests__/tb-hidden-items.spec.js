import { mount } from '@vue/test-utils'
import tbHiddenItems from './../tb-hidden-items'

describe('tb-hidden-items', () => {

    it('renders single ul', () => {

        let wrapper = mount(tbHiddenItems, {
            propsData: {
                data: {},
                hiddenOptions: [],
                index: 0
            }
        })

        expect(wrapper.findAll('ul').length).toEqual(1)
    })

    it('renders basic ul\'s class', () => {

        let wrapper = mount(tbHiddenItems, {
            propsData: {
                data: {},
                hiddenOptions: [],
                index: 0
            }
        })

        expect(wrapper.find('ul').classes()).toEqual(['int-table__hidden-items'])
    })

    it('renders multiple li\'s', () => {

        let wrapper = mount(tbHiddenItems, {
            propsData: {
                data: {
                    one: {},
                    two: {}
                },
                hiddenOptions: [
                    { name: 'one' },
                    { name: 'two' }
                ],
                index: 0
            }
        })

        expect(wrapper.findAll('li').length).toEqual(2)
    })

    it('renders basic li\'s class', () => {

        let wrapper = mount(tbHiddenItems, {
            propsData: {
                data: {
                    one: {},
                    two: {}
                },
                hiddenOptions: [
                    { name: 'one' },
                    { name: 'two' }
                ],
                index: 0
            }
        })

        expect(wrapper.findAll('li').at(0).classes()).toEqual(['int-table__hidden-item'])

        expect(wrapper.findAll('li').at(1).classes()).toEqual(['int-table__hidden-item'])
    })

    it('skips comment nodes (cells, hidden with v-show)', () => {

        let wrapper = mount(tbHiddenItems, {
            propsData: {
                data: {
                    one: { isComment: true },
                    two: {}
                },
                hiddenOptions: [
                    { name: 'one' },
                    { name: 'two' }
                ],
                index: 0
            }
        })

        expect(wrapper.findAll('li').length).toEqual(1)
    })

    it('matches overall snapshot', () => {

        let wrapper = mount(tbHiddenItems, {
            propsData: {
                data: {
                    one: { isComment: true },
                    two: {}
                },
                hiddenOptions: [
                    { name: 'one' },
                    { name: 'two' }
                ],
                index: 0,
                active: true
            }
        })

        expect(wrapper.html()).toMatchSnapshot();
    })
})