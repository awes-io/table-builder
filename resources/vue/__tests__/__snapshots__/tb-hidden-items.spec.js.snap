// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tb-hidden-items matches overall snapshot 1`] = `"<ul class=\\"int-table__hidden-items\\"><li class=\\"int-table__hidden-item\\"></li></ul>"`;
